src/components/organisms/AppRouter/Provided.tsx

```diff
@@ -0,0 +1,44 @@
+import React from "react";
+
+import { useVenueId } from "hooks/useVenueId";
+import { WorldUsersProvider } from "hooks/users";
+import { RelatedVenuesProvider } from "hooks/useRelatedVenues";
+
+interface ProvidedProps {
+  withWorldUsers?: boolean;
+  withRelatedVenues?: boolean;
+}
+
+export const Provided: React.FC<ProvidedProps> = ({
+  withWorldUsers = false,
+  withRelatedVenues = false,
+  children,
+}) => {
+  const venueId = useVenueId();
+
+  if (withWorldUsers && withRelatedVenues) {
+    return (
+      <WorldUsersProvider venueId={venueId}>
+        <RelatedVenuesProvider venueId={venueId}>
+          {children}
+        </RelatedVenuesProvider>
+      </WorldUsersProvider>
+    );
+  }
+
+  if (withWorldUsers) {
+    return (
+      <WorldUsersProvider venueId={venueId}>{children}</WorldUsersProvider>
+    );
+  }
+
+  if (withRelatedVenues) {
+    return (
+      <RelatedVenuesProvider venueId={venueId}>
+        {children}
+      </RelatedVenuesProvider>
+    );
+  }
+
+  return <>{children}</>;
+};
```

Comment:
The logic here can be simplified by making a 'noop' style wrapper component to use in the cases where we don't need them all. I'll make some changes and push them.

Reply:
My original intent was to do some "clever" rebinding of the variable i.e.

```
let wrapper = component;
if(case1)
  wrapper = <Provider>{wrapper}</Provider}
if(case2
// etc.
```

But due to time constraints,, I aborted the attempt and stuck with the dumbest but working solution

Reply:
The 'clever' pattern I was thinking of is basically this: 

https://github.com/sparkletown/sparkle/blob/staging/src/index.tsx#L234-L236

Something like this (untested, but should give the gist):

```typescript
const MaybeWorldUsersProvider = withWorldUsers ? WorldUsersProvider : React.Fragment;
const MaybeRelatedVenuesProvider = withRelatedVenues ? RelatedVenuesProvider : React.Fragment;

return (
  <MaybeWorldUsersProvider venueId={venueId}>
    <MaybeRelatedVenuesProvider venueId={venueId}>
      {children}
    </MaybeRelatedVenuesProvider>
  </MaybeWorldUsersProvider>
);
```

Reply:
fixed in https://github.com/sparkletown/sparkle/pull/1769/commits/a037b8c6644fbb481981e2a84327fe2fe53e0ec9

Mostly followed my plan from https://github.com/sparkletown/sparkle/pull/1769#discussion_r667283279 but added some type safety as well.